Cython/Compiler/FlowControl.py

```diff
@@ -704,8 +706,8 @@ def visit_FuncDefNode(self, node):
         self.visitchildren(node, ('decorators',))
         self.env_stack.append(self.env)
         self.env = node.local_scope
-        self.stack.append(self.flow)
         self.flow = ControlFlow()
+        self.flows[self.env] = self.flow
```

Comment:
Fair point. I'll see if it's possible to clean it up as I go (intuitively it seems like it should be...) and maybe use a tuple-pair stack instead of a dictionary.

Reply:
I've reverted back to a stack so hopefully this concern is gone. I've merged the flow stack and the env stack into one so the env can be used as a "key" though.